scripts/west_commands/runners/nrfutil.py

```diff
@@ -40,27 +42,34 @@ def do_create(cls, cfg, args):
                                    recover=args.recover)
 
     def _exec(self, args):
-        try:
-            out = self.check_output(['nrfutil', '--json', 'device'] + args)
-        except subprocess.CalledProcessError as e:
-            # https://docs.python.org/3/reference/compound_stmts.html#except-clause
-            cpe = e
-            out = cpe.stdout
-        else:
-            cpe = None
-        finally:
-            # https://github.com/ndjson/ndjson-spec
-            out = [json.loads(s) for s in
-                   out.decode(sys.getdefaultencoding()).split('\n') if len(s)]
-            self.logger.debug(f'output: {out}')
-            if cpe:
-                if 'execute-batch' in args:
-                    for o in out:
-                        if o['type'] == 'batch_end' and o['data']['error']:
-                            cpe.returncode = o['data']['error']['code']
-                raise cpe
-
-        return out
+        jout_all = []
+
+        cmd = ['nrfutil', '--json', 'device'] + args
+        self._log_cmd(cmd)
+
+        if _DRY_RUN:
+            return {}
+
+        with subprocess.Popen(cmd, stdout=subprocess.PIPE) as p:
+            for line in iter(p.stdout.readline, b''):
```

Comment:
Actually, this would be simpler witt:

`for line in p.stdout:`

Reply:
Although this might not behave the same way when it comes to buffering the stdout, so feel free to ignore this suggestion.